### DISCUSSION TITLE
Error - "This HeaderNode already contains the similar child node." while adding ANT.

### DISCUSSION THREAD

**Discussion by LokeshNagireddy** [actions/virtual-environments#9846] [General questions]
Hi Team,

Im getting below error with latest build reference from May10th.

![image](https://github.com/actions/runner-images/assets/113025693/34eb63f3-22b2-435e-9aa5-edd863d9c137)

Can you please check and solve the issue. Please let me know if we need to create a bug for this.

image reference: 441122013e49b8a909c8007712da07504a2b4937

We look forward for your response.

**Comment by erik-bershel** [comment]
Hey @LokeshNagireddy!

Latest scheduled nightly builds:
- [uburnut-20.04](https://dev.azure.com/github/runner-images/_build/results?buildId=135699&view=results);
- [ubuntu-22.04](https://dev.azure.com/github/runner-images/_build/results?buildId=135697&view=results);
- [windows-2019](https://dev.azure.com/github/runner-images/_build/results?buildId=135696&view=results);
- [windows-2022](https://dev.azure.com/github/runner-images/_build/results?buildId=135698&view=results);
- [macos-12](https://dev.azure.com/github/runner-images/_build/results?buildId=135695&view=results).

Make sure that your fork is synchroniыed with the current latest state of the `main` branch of our repository, make sure that your local changes, if any, will not affect the build. If the problem persists in this case, then please provide the following information:
- what kind of image are you building;
- when the problem arose;
- link to the repository used.

**Comment by LokeshNagireddy** [comment]
Hi @erik-bershel,

our pipeline was successful until cleanup.sh provisioner in the file ubuntu-22.04.pkr.hcl.
Please find the answers below.

1. what kind of image are you building;
Im building a linux image.

2. when the problem arose;
Problem started from this image reference: 441122013e49b8a909c8007712da07504a2b4937 and above.

3. link to the repository used.
https://github.com/actions/runner-images/tree/dca8e74e477ece24d916fe1dd6d14e010f67a8f7

Before this image reference, there are other issues, but all of them solved from reference of May10th and above and ended up with this latest error.

we are using Image Version: 20240422.1.0

please let me know for more info.

**Reply by erik-bershel** [reply]
Ubuntu-22.04 images builds fine both nightly and stable. Latest nightly [example](https://dev.azure.com/github/runner-images/_build/results?buildId=135716&view=results). Issue placed on your side. I see that the problem is localised in the software report module. The same code works normally when I trying to build latest commit locally.

**Reply by LokeshNagireddy** [reply]
Hi @erik-bershel , Yes, you are right. i found a task in our repo that do some checks on the ANT, Gradle etc. Now, i have removed it and that issue on ANT was solved. However, i got another issue on finding Docker Server version.

![image](https://github.com/actions/runner-images/assets/113025693/f8d0f106-6ea9-4242-be44-51b83c833054)

This error occurred when im using the same image reference as before.

can you please let me know what caused the error, i dont see any additional code in our repo as well as any deviation from the source.

i look forward for your response.

**Reply by LokeshNagireddy** [reply]
Hi @erik-bershel ,

My latest build got into a similar error, but for Helm.
![image](https://github.com/actions/runner-images/assets/113025693/23cba4e9-6eed-459f-8482-b5e1f6046ba7)

Not sure, where is the problem. Can you please help.

**Reply by erik-bershel** [reply]
Helm node was broken by latest release. Is in progress now. Might be tracked here: https://github.com/actions/runner-images/issues/9865

**Reply by erik-bershel** [reply]
> i got another issue on finding Docker Server version.

Lets check it after Helm fixed.

**Reply by LokeshNagireddy** [reply]
This issue with docker server was already solved with latest image reference. you can ignore it for now.

**Comment by erik-bershel** [comment]
Hey @LokeshNagireddy!
How it's going now with your build? Blocker patched some time ago.

**Reply by LokeshNagireddy** [reply]
Hi @erik-bershel,

The build issue solved once you have reverted the helm version. Currently our linux build was successful. However, we are getting errors for windows image for install helpers in Ubuntu22.

i will create another discussion with log info.

Thanks alot for your quick turnover.

**Comment by samus-aran** [comment]
We're locking this discussion because it has not had recent activity and/or other members have asked for more information to assist you but received no response. Thank you for helping us maintain a productive and tidy community for all our members.
